src/sentry/static/sentry/app/components/avatar.jsx

```diff
@@ -23,24 +22,41 @@ class Avatar extends React.Component {
   constructor(...args) {
     super(...args);
     this.state = {
+      gravatarUrl: null,
       showBackupAvatar: false,
       loadError: false,
     };
   }
 
+  componentDidMount() {
+    this.buildGravatarUrl();
+  }
+
+  componentWillReceiveProps() {
+    this.buildGravatarUrl();
+  }
+
   buildGravatarUrl = () => {
-    let url = ConfigStore.getConfig().gravatarBaseUrl + '/avatar/';
+    let avatarType = this.getAvatarType();
 
-    url += MD5(this.props.user.email.toLowerCase());
+    if (avatarType !== 'gravatar') return;
 
-    let query = {
-      s: this.props.size || undefined,
-      d: this.props.default || 'blank',
-    };
+    import(/*webpackChunkName: "crypto-js/md5"*/ 'crypto-js/md5').then(MD5 => {
+      let {user} = this.props;
+      let url = ConfigStore.getConfig().gravatarBaseUrl + '/avatar/';
 
-    url += '?' + $.param(query);
+      url += MD5(user.email.toLowerCase());
 
-    return url;
+      let query = {
+        s: this.props.size || undefined,
+        d: this.props.default || 'blank',
+      };
+
+      url += '?' + $.param(query);
+
+      this.setState({gravatarUrl: url});
+      return url;
```

Comment:
Is this `return` statement necessary anymore (since it's now returned by the `import` callback)?

Reply:
No